server/src/test/java/org/elasticsearch/search/internal/ContextIndexSearcherTests.java

```diff
@@ -192,6 +193,16 @@ public void testConcurrentSearchAllThreadsFinish() throws Exception {
         AtomicInteger missingDocs = new AtomicInteger();
         AtomicInteger visitDocs = new AtomicInteger(0);
 
+        // determine how many docs are in the first slice for correct, this is how much we are missing by
+        // throwing the exception in the first collector
+        LeafSlice[] leafSlices = ContextIndexSearcher.computeSlices(
+            directoryReader.getContext().leaves(),
+            executor.getMaximumPoolSize(),
+            1
+        );
+        // The test collector manager throws an exception when the first segment gets collected.
+        // All documents in that slice count towards the "missing" docs in the later assertion.
+        int docsFirstSlice = Arrays.stream(leafSlices[0].leaves).map(LeafReaderContext::reader).mapToInt(LeafReader::maxDoc).sum();
```

Comment:
Thanks for clarifying the comments. I am a bit confused on what part of your change requires adapting this test. Sorry, I must be missing something.

Reply:
The assertion in L271 checks that visitDocs.get() + missingDocs.get(), equalTo(numDocs). So far the slicing overwrite we did here forces one segment for slice. With our own slicing logic now (`computeSlices(List<LeafReaderContext> leaves, int maxSliceNum, int minDocsPerSlice)`) we don't have that guarantee, we often end up with a few smaller segments in the first slice. Thats the one triggering the exception in L217. For the numbers to add up we not only need to count the docs from that one segment as "missing" but all docs in that slice.

Reply:
ok, and what is the value of changing how slices was overridden in this test?

Reply:
I see, thanks, that makes sense.